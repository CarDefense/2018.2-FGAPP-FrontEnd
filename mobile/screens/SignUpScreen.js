import { PROFILE_API } from './tab_navigator/car_defense/screens/TabNavigator/const/Const'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
import { TextField } from 'react-native-material-textfield';
import { Constants, ImagePicker, Permissions } from 'expo';
import React, { Component } from 'react';
import { Icon } from "native-base";
import {
  ActivityIndicator,
  TouchableOpacity,
  View,
  StatusBar,
  Text,
  Alert,
  StyleSheet,
  FlatList,
  ImageBackground,
  Image,
  ScrollView,
  KeyboardAvoidingView
} from 'react-native';

async function _Alert(){
  Alert.alert('Atenção', 'Recomenda-se o envio de foto de um documento de identificação com foto durante o cadastro.')
}

export default class SignUpScreen extends Component {
  state = {
    loading: true, 
  }

  componentWillMount() {
    _Alert();
  }

  constructor(props) {
    super(props);

    this.onFocus = this.onFocus.bind(this);
    //this.onSubmit = this.onSubmit.bind(this);
    //this.onChangeText = this.onChangeText.bind(this);
    this.onSubmitEmail = this.onSubmitEmail.bind(this);
    this.onSubmitUsername = this.onSubmitUsername.bind(this);
    this.onSubmitPassword = this.onSubmitPassword.bind(this);
    this.onAccessoryPress = this.onAccessoryPress.bind(this);

    this.emailRef = this.updateRef.bind(this, 'email');
    this.usernameRef = this.updateRef.bind(this, 'username');
    this.passwordRef = this.updateRef.bind(this, 'password');

    this.renderPasswordAccessory = this.renderPasswordAccessory.bind(this);

    this.state = {
      email: '',
      userName: '',
      password: '',
      secureTextEntry: true,
      username_field_alerts: [''],
      uploading: false,
    };

  }

  onFocus() {
    let { errors = {} } = this.state;

    for (let name in errors) {
      let ref = this[name];

      if (ref && ref.isFocused()) {
        delete errors[name];
      }
    }

    this.setState({ errors });
  }

  onChangeText(text) {
    ['email', 'username', 'password']
      .map((name) => ({ name, ref: this[name] }))
      .forEach(({ name, ref }) => {
        if (ref.isFocused()) {
          this.setState({ [name]: text });
        }
      });
  }

  onAccessoryPress() {
    this.setState(({ secureTextEntry }) => ({ secureTextEntry: !secureTextEntry }));
  }

  onSubmitEmail() {
    this.username.focus();
  }

  onSubmitUsername() {
    this.password.focus();
  }

  onSubmitPassword() {
    this.password.blur();
  }

  updateRef(name, ref) {
    this[name] = ref;
  }

  renderPasswordAccessory() {
    let { secureTextEntry } = this.state;

    let name = secureTextEntry ?
      'visibility' :
      'visibility-off';

    return (
      <MaterialIcon
        size={24}
        name={name}
        color={TextField.defaultProps.baseColor}
        onPress={this.onAccessoryPress}
        suppressHighlighting
      />
    );
  }

  _onPressButton = async () => {
    let errors = {};
    let errorEmail = false;
    let errorUserName = false;
    let errorPassword = false;

    ['email']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Digite o email';
          errorEmail = true;
        }
        else {
          if(!value.includes("@") || !value.includes(".com")){
            errors[text] = 'Digite um email válido';
            errorEmail = true;
          }
        }
      });

    ['username']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Digite o nome de usuário';
          errorUserName = true;
        }
      });

    ['password']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Digite uma senha';
          errorPassword = true;
        }
        else {
          if (value.length < 8) {
            errors[text] = 'Use 8 caracteres ou mais para a sua senha';
            errorPassword = true;
          }
          else if (value.length > 15) {
            errors[text] = 'Use 15 caracteres ou menos para a sua senha';
            errorPassword = true;
          }
        }
      });

    if (!errorUserName && !errorPassword && !errorEmail) {
      this.setState({
        loading: true
      });

      var registration_path = PROFILE_API + '/users/';
      fetch(registration_path, {
        method: 'POST',
        credentials: 'include',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          'username': this.state.username,
          'password': this.state.password
        }),
      })
        .then((response) => response.json())
        .then((responseJson) => {
          console.log(JSON.stringify(responseJson));
          if (responseJson.username != undefined) {
            this.setState({ username_field_alerts: ['Um usuário já foi cadastrado com esse nome.'] })
            this.setState({
              loading: false
          });
          }
          else {
            this.setState({ username_field_alerts: [''] })
          }
          if (responseJson.token != undefined ||
            responseJson.key != undefined) {
              this.setState({
                loading: false
            });
            Alert.alert("Conta criada com sucesso!");
            this.props.navigation.navigate('WelcomeScreen')
          }
        })

        .catch(err => {
          if (typeof err.text === 'function') {
            this.setState({
              loading: false
          });
            err.text().then(errorMessage => {
              this.props.dispatch(displayTheError(errorMessage))
            });
          } else {
            Alert.alert("Erro na conexão.");
            console.log(err)
            this.setState({
              loading: false
          });
          }
        });
    }
    this.setState({ errors });
  }

  render() {
    let { errors = {}, secureTextEntry, ...data } = this.state;
    
    if(!this.state.loading){
      return (
        <ImageBackground
          source={require('../images/b6.jpg')}
          style={{ width: '100%', height: '100%' }}
          >
          <KeyboardAvoidingView behavior="position">
            <ScrollView>
              <View style={styles.container}>
                <View style={styles.containerImage}>
                  <Image
                    style={styles.image}
                    source={require('../images/icontest.png')}
                    />
                </View>
                <View style={styles.container1}>
                  <TextField
                    ref={this.emailRef}
                    value={data.email}
                    keyboardType='email-address'
                    autoCapitalize='none'
                    autoCorrect={false}
                    enablesReturnKeyAutomatically={true}
                    onFocus={this.onFocus}
                    onChangeText={(email) => this.setState({ email })}
                    onSubmitEditing={this.onSubmitEmail}
                    returnKeyType='next'
                    label='Endereço de email'
                    tintColor="white"
                    underlineColorAndroid="transparent"
                    error={errors.email}
                    textColor='white'
                    />
                  <TextField
                    ref={this.usernameRef}
                    value={data.username}
                    autoCorrect={false}
                    enablesReturnKeyAutomatically={true}
                    onFocus={this.onFocus}
                    onChangeText={(username) => this.setState({ username })}
                    onSubmitEditing={this.onSubmitUsername}
                    returnKeyType='next'
                    label='Nome de usuário'
                    tintColor="white"
                    underlineColorAndroid="transparent"
                    error={errors.username}
                    textColor='white'
                    />
                  <TextField
                    ref={this.passwordRef}
                    value={data.password}
                    secureTextEntry={secureTextEntry}
                    autoCapitalize='none'
                    autoCorrect={false}
                    enablesReturnKeyAutomatically={true}
                    clearTextOnFocus={true}
                    onFocus={this.onFocus}
                    onChangeText={(password) => this.setState({ password })}
                    onSubmitEditing={this.onSubmitUsername}
                    returnKeyType='done'
                    label='Senha'
                    tintColor="white"
                    underlineColorAndroid="transparent"
                    error={errors.password}
                    maxLength={20}
                    characterRestriction={15}
                    renderAccessory={this.renderPasswordAccessory}
                    textColor='white'
                    />
                  <View style={styles.container2}>
                    <TouchableOpacity
                      color="#B2EBF2"
                      onPress={this._takePhoto}
                      containerViewStyle={{ width: '10%' }}
                      >
                      <Icon
                        type='FontAwesome'
                        name="camera"
                        style={{ color: "white" }}
                        />
                    </TouchableOpacity>
                  </View>
                  {this._maybeRenderImage()}
                  {this._maybeRenderUploadingOverlay()}
                  <View style={styles.containerButton}>
                    <TouchableOpacity
                      style={styles.button}
                      onPress={() => this._onPressButton()}
                      containerViewStyle={{ width: '40%' }}
                      >
                      <Text style={{ color: 'white' }} >Cadastrar</Text>
                    </TouchableOpacity>
                    <FlatList
                      data={this.state.username_field_alerts}
                      renderItem={({ item }) => <Text style={{ color: 'red' }}>{item}</Text>}
                      keyExtractor={item => 'non_field_errors'}
                      />
                  </View>
                </View>
              </View>
            </ScrollView>
          </KeyboardAvoidingView>
        </ImageBackground>
      );
    } else {
      return (
        <ImageBackground
          source={require('../images/b6.jpg')}
          style={{ width: '100%', height: '100%' }}
          >
          <ScrollView>
            <View style={styles.container}>
              <View style={styles.containerImage}>
                <Image
                  style={styles.image}
                  source={require('../images/icontest.png')}
                  />
              </View>
              <View style={styles.container3}>
                <View
                  style={[StyleSheet.absoluteFill, styles.maybeLoading]}>
                  <ActivityIndicator color="#ffffff" size="large" />
                </View>
              </View>
            </View>
          </ScrollView>
        </ImageBackground>
      ); 
    }
  }

  _maybeRenderUploadingOverlay = () => {
    if (this.state.uploading) {
      return (
        <View
          style={styles.maybeRenderUploading}>
          <ActivityIndicator color="#313869" size="large" />
        </View>
      );
    }
  };

  _maybeRenderImage = () => {
    let {
      image
    } = this.state;

    if (!image) {
      return;
    }

    // onPress={this._copyToClipboard}
    // onLongPress={this._share}

    return (
      <View
        style={styles.maybeRenderContainer}>
        <View
          style={styles.maybeRenderImageContainer}>
          <Image source={{ uri: image }} style={styles.maybeRenderImage} />
        </View>
      </View>
    );
  };

  _share = () => {
    Share.share({
      message: this.state.image,
      title: 'Check out this photo',
      url: this.state.image,
    });
  };

  _copyToClipboard = () => {
    Clipboard.setString(this.state.image);
    alert('Copied image URL to clipboard');
  };

  _takePhoto = async () => {
    const {
      status: cameraPerm
    } = await Permissions.askAsync(Permissions.CAMERA);

    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera AND camera roll
    if (cameraPerm === 'granted' && cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchCameraAsync({
        allowsEditing: true,
      });

      this._handleImagePicked(pickerResult);
    }
  };

  _pickImage = async () => {
    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera roll
    if (cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchImageLibraryAsync({
        allowsEditing: true,

      });

      this._handleImagePicked(pickerResult);
    }
  };

  _handleImagePicked = async pickerResult => {
    let uploadResponse, uploadResult;

    try {
      this.setState({
        uploading: true
      });

      if (!pickerResult.cancelled) {
        uploadResponse = await uploadImageAsync(pickerResult.uri);
        uploadResult = await uploadResponse.json();

        console.log(uploadResult.image)

        this.setState({
          image: uploadResult.image
        });
      }
    } catch (e) {
      console.log({ uploadResponse });
      console.log({ uploadResult });
      console.log({ e });
      alert('Falha de Upload, tente novamente.');
    } finally {
      this.setState({
        uploading: false
      });
    }
  };
}

async function uploadImageAsync(uri) {
  let apiUrl = PROFILE_API + '/documents/';

  let uriParts = uri.split('.');
  let fileType = uriParts[uriParts.length - 1];

  let formData = new FormData();
  formData.append('image', {
    uri,
    name: `photo.${fileType}`,
    type: `image/${fileType}`,
  });

  console.log(uri)

  let options = {
    method: 'POST',
    body: formData,
    headers: {
      Accept: 'application/json',
      'Content-Type': 'multipart/form-data',
    },
  };
  return fetch(apiUrl, options);
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
    flex: 1
  },
  containerImage: {
    margin: 30,
    paddingTop: 4,
    paddingRight: 4,
    alignItems: 'center'
  },
  image: {
    height: 100,
    width: 100,
  },
  container1:{
    marginTop: 70,
    marginBottom: 12
  },
  container2: {
    marginTop: 8,
    flexDirection: 'row',
    justifyContent: 'center',
    bottom: 10
  },
  containerButton: {
    alignItems: 'center',
    marginTop: 6
  },
  button: {
    backgroundColor: "#26C6DA",
    borderRadius: 15,
    height: 40,
    width: 325,
    justifyContent: 'center',
    alignItems: 'center'
  },
  container3: {
    paddingTop: "50%",
  },
  maybeLoading: {
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0)',
    justifyContent: 'center',
  },
  maybeRenderUploading: {
    paddingTop: 8,
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.0)',
    justifyContent: 'center',
  },
  maybeRenderContainer: {
    width: '100%',
    height: '40%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    elevation: 2,
    marginTop: 32,
    shadowColor: 'rgba(0,0,0,1)',
    shadowOpacity: 0.2,
    shadowOffset: {
      height: 4,
      width: 4,
    },
    shadowRadius: 5
  },
  maybeRenderImageContainer: {
    borderTopLeftRadius: 3,
    borderTopRightRadius: 3,
    overflow: 'hidden',
  },
  maybeRenderImage: {
    height: 250,
    width: 250,
  },

  containerText: {
    margin: 60,
    alignItems: 'center'
  },
});
