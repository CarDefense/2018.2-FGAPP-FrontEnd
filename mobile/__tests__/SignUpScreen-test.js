import SignUpScreen from '../screens/SignUpScreen'
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

test('SignUpScreen snapShot', () => {
    const snap = renderer.create(
        <SignUpScreen/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})