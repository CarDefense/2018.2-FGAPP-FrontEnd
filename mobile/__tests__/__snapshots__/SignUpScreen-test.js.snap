// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen snapShot 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={undefined}
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "bottom": 0,
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "margin": 30,
                  "paddingRight": 4,
                  "paddingTop": 4,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 100,
                    "width": 100,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginBottom": 12,
                  "marginTop": 70,
                }
              }
            >
              <View
                onResponderRelease={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="auto"
                style={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                      "borderBottomWidth": 0.5,
                      "height": 64,
                      "paddingBottom": 8,
                      "paddingTop": 32,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "position": "absolute",
                        "top": 36,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.38)",
                          "fontSize": 16,
                        }
                      }
                    >
                      Endereço de email
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="none"
                      autoCorrect={false}
                      disableFullscreenUI={true}
                      editable={true}
                      enablesReturnKeyAutomatically={true}
                      error={undefined}
                      keyboardType="email-address"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onChangeText={[Function]}
                      onContentSizeChange={[Function]}
                      onFocus={[Function]}
                      onSubmitEditing={[Function]}
                      returnKeyType="next"
                      selectionColor="white"
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "margin": 0,
                            "padding": 0,
                            "top": 2,
                          },
                          Object {
                            "color": "white",
                            "fontSize": 16,
                            "height": 24,
                            "textAlign": "left",
                          },
                          undefined,
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                </View>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 8,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgb(213, 0, 0)",
                            "fontSize": 0,
                            "opacity": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgba(0, 0, 0, .38)",
                            "fontSize": 12,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                onResponderRelease={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="auto"
                style={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                      "borderBottomWidth": 0.5,
                      "height": 64,
                      "paddingBottom": 8,
                      "paddingTop": 32,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "position": "absolute",
                        "top": 36,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.38)",
                          "fontSize": 16,
                        }
                      }
                    >
                      Nome de usuário
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="sentences"
                      autoCorrect={false}
                      disableFullscreenUI={true}
                      editable={true}
                      enablesReturnKeyAutomatically={true}
                      error={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onChangeText={[Function]}
                      onContentSizeChange={[Function]}
                      onFocus={[Function]}
                      onSubmitEditing={[Function]}
                      returnKeyType="next"
                      selectionColor="white"
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "margin": 0,
                            "padding": 0,
                            "top": 2,
                          },
                          Object {
                            "color": "white",
                            "fontSize": 16,
                            "height": 24,
                            "textAlign": "left",
                          },
                          undefined,
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                </View>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 8,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgb(213, 0, 0)",
                            "fontSize": 0,
                            "opacity": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgba(0, 0, 0, .38)",
                            "fontSize": 12,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                onResponderRelease={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="auto"
                style={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                      "borderBottomWidth": 0.5,
                      "height": 64,
                      "paddingBottom": 8,
                      "paddingTop": 32,
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "position": "absolute",
                        "top": 36,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      collapsable={undefined}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.38)",
                          "fontSize": 16,
                        }
                      }
                    >
                      Senha
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="none"
                      autoCorrect={false}
                      disableFullscreenUI={true}
                      editable={true}
                      enablesReturnKeyAutomatically={true}
                      error={undefined}
                      maxLength={20}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onChangeText={[Function]}
                      onContentSizeChange={[Function]}
                      onFocus={[Function]}
                      onSubmitEditing={[Function]}
                      renderAccessory={[Function]}
                      returnKeyType="done"
                      secureTextEntry={true}
                      selectionColor="white"
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "margin": 0,
                            "padding": 0,
                            "top": 2,
                          },
                          Object {
                            "color": "white",
                            "fontSize": 16,
                            "height": 24,
                            "textAlign": "left",
                          },
                          undefined,
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    />
                    <View
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "justifyContent": "center",
                          "top": 2,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                      />
                    </View>
                  </View>
                </View>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 24,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgb(213, 0, 0)",
                            "fontSize": 0,
                            "opacity": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgba(0, 0, 0, .38)",
                            "fontSize": 12,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "paddingLeft": 4,
                        "paddingVertical": 4,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "textAlign": "right",
                          },
                          undefined,
                          Object {
                            "color": "rgba(0, 0, 0, .38)",
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      0
                       / 
                      15
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "bottom": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "marginTop": 6,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#26C6DA",
                      "borderRadius": 15,
                      "height": 40,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 325,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    Cadastrar
                  </Text>
                </View>
                <RCTScrollView
                  data={
                    Array [
                      "",
                    ]
                  }
                  disableVirtualization={false}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  horizontal={false}
                  initialNumToRender={10}
                  invertStickyHeaders={undefined}
                  keyExtractor={[Function]}
                  maxToRenderPerBatch={10}
                  numColumns={1}
                  onContentSizeChange={[Function]}
                  onEndReachedThreshold={2}
                  onLayout={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  updateCellsBatchingPeriod={50}
                  viewabilityConfigCallbackPairs={Array []}
                  windowSize={21}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "red",
                          }
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
